docs/guide/template-filters.md

```diff
@@ -0,0 +1,131 @@
+# Template filters
+
+## Nunjucks supplied filters
+The Nunjucks templating language comes with several [built-in filters](https://mozilla.github.io/nunjucks/templating.html#filters). These filters apply functions to template data before outputting content to the page. They are called with a pipe operator `|` and can take arguments.
+
+```markup
+<h1>{{ data.page.headline | replace("foo", "bar" ) | upper }}</h1>
+```
+
+In this example, a hypothetical page headline will be retrieved from the page data and then piped to the 'replace' filter, along with two arguments. This filter will scan the incoming data and replace any instances of ‘foo’ with 'bar'. The output of this filter will then pass the result to the upper function, which will return the input as an uppercase string before being output to the page.
+
+>Note: The order of filters can be significant, as they are applied sequentially to the input data.
+
+## Apostrophe supplied filters
+
+Several custom filters and filter sets (multiple, sequential filters) are used in Apostrophe templating. Rather than remembering each filter and the correct order, apostrophe exposes them automatically to the templates. In addition, Apostrophe also adds multiple "helper functions" which do not use the `| foo` filter syntax. The most common of these are described in the [alphabetical filter reference](/guide/template-filters.html#alphabetical-apostrophe-filter-reference).
+
+## Custom template filters
+
+If you have special template needs you can also construct custom filters for use in Nunjucks templates. They can be added by passing the filter name and function to the template module `addFilter(name, function)` method. Multiple filters can be added by passing an object to this same method, where each key is the name of the filter and the value is the function.
+
+By way of example, we'll create a link where the URL is included as part of the label, but without the protocol.
+```javascript
```

Comment:
Check out the `AposCodeBlock` component used elsewhere for fancier and consistent styling and file name placement.

Reply:
Thanks - I will look for this in the future when I hit code blocks. I think there are some docs I already added without that formatting.